kolibri/core/assets/src/views/sync/SelectAddressModalGroup/SelectAddressForm.vue

```diff
@@ -295,63 +323,120 @@
           return `${url}, Kolibri ${version}`;
         } else return url;
       },
-      resetSelectedAddress() {
-        if (this.availableAddressIds.length !== 0) {
-          const selectedId = this.selectedId || this.storageAddressId || this.selectedAddressId;
-          this.selectedAddressId =
-            this.availableAddressIds.find(id => id === selectedId) || this.availableAddressIds[0];
+      resetSelectedDevice() {
+        if (this.availableDeviceIds.length !== 0) {
+          const selectedId = this.selectedId || this.storageDeviceId || this.selectedDeviceId;
+          this.selectedDeviceId =
+            this.availableDeviceIds.find(id => id === selectedId) || this.availableDeviceIds[0];
         } else {
-          this.selectedAddressId = '';
+          this.selectedDeviceId = '';
         }
       },
       handleSubmit() {
-        if (this.selectedAddressId) {
-          const match = find(this.combinedAddresses, { id: this.selectedAddressId });
-          match.isDynamic = Boolean(find(this.discoveredAddresses, { id: this.selectedAddressId }));
-          this.$emit('submit', match);
+        if (!this.selectedDeviceId) {
+          return;
         }
+
+        const match = find(this.devices, { id: this.selectedDeviceId });
+        if (!match) {
+          this.uiAlertText = this.$tr('genericConnectionError');
+          return this.forceFetch();
+        }
+
+        this.uiAlertText = null;
+        this.isSubmitChecking = true;
+        this.doCheck(match.id).then(device => {
+          if (device.available) {
+            this.$emit('submit', device);
+          } else {
+            this.isSubmitChecking = false;
+            switch (device.connection_status) {
+              case ConnectionStatus.Unknown:
```

Comment:
As `break` does not exist, this will go to the `default` leg of the switch, not sure if this is done on purpose

Reply:
It doesn't actually go to the `default` leg, it gets combined with the next switch for `ConnectionStatus.ConnectionFailure`

Reply:
can `device.connection_status` be `ConnectionStatus.Unknown` & `ConnectionStatus.ConnectionFailure ` at the same time? If not I only see the `default` leg as possible without the `break`. If it can, I don't see the need to query for `ConnectionStatus.Unknown`
In summary, I don't feel comfortable removing the `break` statement inside a `switch`, it removes the readability that switch adds over a series of chained `if-else` and there are usually cleaner options.

Reply:
In this situation, it becomes an OR statement on either `ConnectionStatus.Unknown` or `ConnectionStatus.ConnectionFailure`:
```
> let x = 'a';
undefined
> switch (x) {
... case 'a':
... case 'b':
...   console.log('case a or b')
...   break;
... default:
...   console.log('no case')
... }
case a or b
```
This type of situation is exactly why would tend to use a `switch`, because you can map multiple values to a single value with more clarity than a typical block of multiple `if` statements